server/lib/SignatureHandler.php

```diff
@@ -97,4 +98,33 @@ private function splitCloudId(string $cloudId): array {
 		return [$user, $host];
 	}
 
+
+	/**
+	 * @param Request $request
+	 *
+	 * @throws SignedRequestException
+	 */
+	public function verifyRequest(Request $request): string {
+		$body = json_decode($request->getBody(), true);
+		if ($body === null || !isset($body['message']) || !isset($body['message']['data'])
+			|| !isset($body['message']['data']['federationId'])
+			|| !isset($body['signature'])
+			|| !isset($body['message']['timestamp'])) {
+			throw new SignedRequestException();
+		}
+
+		$cloudId = $body['message']['data']['federationId'];
+
+		try {
+			$verified = $this->verify($cloudId, $body['message'], $body['signature']);
+			if ($verified) {
+				list(, $host) = $this->splitCloudId($body['message']['data']['federationId']);
```

Comment:
```suggestion
				[, $host] = $this->splitCloudId($body['message']['data']['federationId']);
```
For consistency with Nextcloud code